app/models/good_job/job.rb

```diff
@@ -761,6 +762,7 @@ def reset_batch_values(&block)
 
     def continue_discard_or_finish_batch
       batch._continue_discard_or_finish(self) if batch.present?
+      callback_batch._continue_discard_or_finish(self) if callback_batch.present?
```

Comment:
This shouldn't pass the job. If the job is a callback_job, it shouldn't have the ability to discard the batch.